tests/integration/test_wazuh_db/test_wazuh_db.py

```diff
@@ -64,79 +72,156 @@ def clean_registered_agents():
     time.sleep(5)
 
 
-@pytest.fixture(scope="module")
-def restart_wazuh():
-    control_service('restart')
-
-
-@pytest.fixture(scope="function")
+@pytest.fixture(scope='module')
 def pre_insert_agents():
+    """Insert agents. Only used for the global queries"""
     AGENTS_CANT = 14000
     AGENTS_OFFSET = 20
     for id in range(AGENTS_OFFSET, AGENTS_OFFSET + AGENTS_CANT):
-        command = f'global insert-agent {{"id":{id},"name":"TestName{id}","date_add":1599223378}}'
-        receiver_sockets[0].send(command, size=True)
-        response = receiver_sockets[0].receive(size=True).decode()
-        data = response.split(" ", 1)
-        assert data[0] == 'ok', f'Unable to add agent {id}'
+        insert_agent(id)
+
+    yield
+
+    for id in range(AGENTS_OFFSET, AGENTS_OFFSET + AGENTS_CANT):
+        remove_agent(id)
+
+
+@pytest.fixture(scope='function')
+def insert_agents_test():
+    """Insert agents. Only used for the agent queries"""
+    agent_list = [1, 2]
+    for agent in agent_list:
+        insert_agent(agent)
+
+    yield
+
+    for agent in agent_list:
+        remove_agent(agent)
+
+
+@pytest.fixture(scope='module')
+def stop_all_daemons(request):
+    yield 
+    control_service('stop')
```

Comment:
In order to avoid possible errors if wazuh is not running at the beginning of the wazuhdb tests, we should include use this fixture instead.
```suggestion
def restart_wazuh(request):
    control_service('start')

    yield 
  
    delete_dbs()
    control_service('stop')
```

This should be refactored in the future in order to use `daemon_handler` and remove split the logic of `configure_sockets_environment` fixture.